src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,367 @@
+        LOG.info(String.format("queries: %s,\tNOT found", Arrays.asList(queries), index));
+      }
+    }
+
+    //Close the input stream
+    br.close();
+
+    LOG.info("Found: " + found);
+    LOG.info("Not Found: " + notfound);
+    double percent = (found * 100.0) / (found + notfound);
+    LOG.info("Found/Total %: " + percent);
+    LOG.info("Querying completed.");
+  }
+
+  /**
+   * Prints query results to the standard output stream.
+   *
+   * @param queryName the entity name to search
+   * @throws Exception on error
+   * @return a list of top ranked entities
+   */
+  public List<RankedEntity> search(String queryName) throws Exception {
+    // Initialize index searcher
+    IndexSearcher searcher = new IndexSearcher(reader);
+    Similarity similarity = new ClassicSimilarity();
+    searcher.setSimilarity(similarity);
+
+    MultiFieldQueryParser queryParser = new MultiFieldQueryParser(
+            new String[]{ IndexTopics.FIELD_NAME,
+                    IndexTopics.FIELD_LABEL },
+            new SimpleAnalyzer());
+    queryParser.setDefaultOperator(QueryParser.Operator.AND);
+    Query query = queryParser.parse(queryName);
+
+    TopDocs rs = searcher.search(query, numHits);
+    ScoredDocuments docs = ScoredDocuments.fromTopDocs(rs, searcher);
+    List<RankedEntity> rankedEntities = new ArrayList<>(numHits);
+
+    for (int i = 0; i < docs.documents.length; i++) {
+      float score = docs.scores[i];
+      String mid = docs.documents[i].getField(IndexTopics.FIELD_TOPIC_MID).stringValue();
+      String shortMid = getShortMid(mid);
+      String name = docs.documents[i].getField(IndexTopics.FIELD_NAME).stringValue();
+      String label = docs.documents[i].getField(IndexTopics.FIELD_LABEL).stringValue();
+      rankedEntities.add(new RankedEntity(shortMid, score, name, label));
+    }
+
+    return rankedEntities;
+  }
+
+  // Removes '<', '>' if they exist, lower case
+  private static String cleanUri(String uri) {
+    if (uri.charAt(0) == '<')
+      return uri.substring(1, uri.length() - 1).toLowerCase();
+    else
+      return uri;
+  }
+
+  // get 1-gram, 2-gram, 3-gram, 4-gram combinations from the text
+  private static List<String> getSearchQueries(String text) {
+    List<String> ngrams = new ArrayList<String>();
+    if (text.endsWith(".") || text.endsWith("?")) {
+      text = text.substring(0, text.length()-1);
+    }
+    String[] tokens = text.split("\\s+");
+
+    // add unigram
+    for (int i=0; i < tokens.length; i++) {
+      ngrams.add(tokens[i]);
+    }
+
+    // add bigram
+    for (int i=1; i < tokens.length; i++) {
+      ngrams.add(tokens[i-1] + " " + tokens[i]);
+    }
+
+    // add 3-gram
+    for (int i=2; i < tokens.length; i++) {
+      ngrams.add(tokens[i-2] + " " + tokens[i-1] + " " + tokens[i]);
+    }
+
+    // add 4-gram
+    for (int i=3; i < tokens.length; i++) {
+      ngrams.add(tokens[i-3] + " " + tokens[i-2] + " " + tokens[i-1] + " " + tokens[i]);
+    }
+
+    return ngrams;
```

Comment:
i added space between = but not - (makes the lines too long and not easy to read)